### DISCUSSION TITLE
root is not authorized to perform

### DISCUSSION THREAD

**Discussion by Danishkar** [aws/aws-sdk-java-v2#5250] [Q&A] [answered]
i am using the implementation 'software.amazon.awssdk:devicefarm:2.25.60' to preform a createUpload to upload my application apk why do i get the below error message
```
 [Request processing failed: software.amazon.awssdk.services.devicefarm.model.DeviceFarmException: User: arn:aws:iam::905418274966:root is not authorized to perform: devicefarm:CreateUpload on resource:
```

**Comment by debora-ito** [comment, marked as answer]
See Troubleshooting - I am not authorized to perform an action in Device Farm
https://docs.aws.amazon.com/devicefarm/latest/developerguide/security_iam_troubleshoot.html
